src/components/cc-icon/cc-icon.js

```diff
@@ -0,0 +1,109 @@
+import { css, html, LitElement } from 'lit';
+import { unsafeSVG } from 'lit/directives/unsafe-svg.js';
+
+/**
+ * @typedef {import('./cc-icon.types.js').IconModel} IconModel
+ * @typedef {import('./cc-icon.types.js').IconSize} IconSize
+ */
+
+/**
+ * A component rendering an SVG icon.
+ *
+ * @cssdisplay inline-flex
+ *
+ * @cssprop {Color} --cc-icon-color - Sets the value of the color CSS property (defaults: `currentColor`).
+ * @cssprop {Size} --cc-icon-size - Sets the value of the width and height CSS properties (defaults: `16px`).
+ */
+export class CcIcon extends LitElement {
+  static get properties () {
+    return {
+      accessibleName: { type: String, attribute: 'accessible-name' },
+      icon: { type: Object },
+      size: { type: String, reflect: true },
+    };
+  }
+
+  constructor () {
+    super();
+
+    /** @type {string|null} Forces the values of the `aria-hidden`, `aria-label` and `role` attributes on the `svg` element. Also add or removes a `<title>` element in the SVG. */
```

Comment:
I think we could make an exception to how we typically write our prop docs here to provide more guidance and a warning about this prop usage:

![image](https://user-images.githubusercontent.com/100240294/205882871-9661a3ce-398f-4092-a547-e1692e5923a3.png)
```suggestion
    /** @type {string|null} Only use this prop if your icon provides information that is not already given in its surrounding text.
     *
     * If this prop has a value:
     *
     *  * sets a `role="img"` and `aria-label="the value of this prop"` attributes on the `<svg>` element.
     *  * creates a `<title>` element inside the `<svg>`.
     *
     * This allows assistive technologies to recognize the element as an image and convey its information.
     *
     * If this prop has no value, sets an `aria-hidden="true"` attribute on the `<svg>` element so that it can be ignored by assistive technologies.
     */
```